src/python/pants/base/exception_sink.py

```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+import logging
+import os
+import sys
+from builtins import object
+
+from pants.util.dirutil import is_writable_dir, safe_open
+
+
+logger = logging.getLogger(__name__)
+
+
+class ExceptionSink(object):
+  """A mutable singleton object representing where exceptions should be logged to."""
+
+  _destination = os.getcwd()
+
+  def __new__(cls, *args, **kwargs):
+    raise TypeError('Instances of {} are not allowed to be constructed!'
+                    .format(cls.__name__))
+
+  class ExceptionSinkError(Exception): pass
+
+  @classmethod
+  def set_destination(cls, dir_path):
+    if not is_writable_dir(dir_path):
+      # TODO: when this class sets up excepthooks, raising this should be safe, because we always
```

Comment:
This could just be a note.

Reply:
I can see you've removed it in #6539, sounds great.

Reply:
If you need any of this in your next PR, let me know and I can backport it. Or feel free to just cherry-pick to your branch.

Reply:
Alternatively, I could land #6539 with the test marked `xfail`.

Reply:
I'll cherry-pick whatever I need, I was thinking -- the changes are mostly orthogonal to what I have now. Landing with it marked xfail is a-ok to me as long as other testing is ok.